unit_tests/memory_monitor/test_memory_monitor_main.cpp

```diff
@@ -0,0 +1,345 @@
+}
+
+void mmon_sub_stat (char *ptr)
+{
+  test_mmon_Gl->sub_stat (ptr);
+}
+
+void mmon_aggregate_server_info (MMON_SERVER_INFO &server_info)
+{
+  test_mmon_Gl->aggregate_server_info (server_info);
+}
+// Dummy APIs end
+
+// Find target stat
+int find_test_stat (MMON_SERVER_INFO &server_info, std::string target)
+{
+  int cnt = 0;
+  for (const auto &[stat_name, mem_usage] : server_info.stat_info)
+    {
+      if (stat_name.compare (target) == 0)
+	{
+	  return cnt;
+	}
+      cnt++;
+    }
+
+  // there is no matching target
+  return -1;
+}
+
+/* ***IMPORTANT!!***
+ * The unit tests for the 'memory_monitor' class are intertwined within one cohesive framework.
+ * Each test depends on all the previous tests. These tests can be broadly categorized into
+ * two categories: 'add' and 'sub'. When developers add new tests,
+ * they must consider the entire test suite.
+ */
+
+// Test add_stat()
+//
+// Test add_stat() with single thread when different file paths
+// that will be made different stat name are given
+//
+// Expected Result: stat name "add_test.c:100" with value "allocated_size" and
+//                  "base/add_test.c" with value "allocated_size_2"
+TEST_CASE ("Test mmon_add_stat", "")
+{
+  MMON_SERVER_INFO test_server_info;
+  size_t allocated_size;
+  size_t allocated_size_2;
+  int ret;
+  REQUIRE (mmon_is_memory_monitor_enabled () == false);
+  mmon_initialize ("unittest");
+  REQUIRE (mmon_is_memory_monitor_enabled () == true);
+
+  mmon_aggregate_server_info (test_server_info);
+  ret = find_test_stat (test_server_info, "add_test.c:100");
+  REQUIRE (ret == -1);
+  test_server_info.stat_info.clear();
+
+  test_mem_in_the_scope_normal = (char *) malloc (32);
+  test_mem_in_the_scope_with_base = (char *) malloc (20 + MMON_METAINFO_SIZE);
+  allocated_size = malloc_usable_size (test_mem_in_the_scope_normal);
+  mmon_add_stat (test_mem_in_the_scope_normal, allocated_size, "/src/add_test.c", 100);
+  allocated_size_2 = malloc_usable_size (test_mem_in_the_scope_with_base);
+  mmon_add_stat (test_mem_in_the_scope_with_base, allocated_size_2, "/src/base/add_test.c", 100);
+
+  mmon_aggregate_server_info (test_server_info);
+  ret = find_test_stat (test_server_info, "add_test.c:100");
+
+  REQUIRE (ret != -1);
+  REQUIRE (test_server_info.total_mem_usage == allocated_size + allocated_size_2);
+  REQUIRE (test_server_info.total_metainfo_mem_usage == MMON_METAINFO_SIZE * 2);
+  REQUIRE (test_server_info.num_stat == 2);
+  REQUIRE (test_server_info.stat_info[ret].second == allocated_size);
+
+  ret = find_test_stat (test_server_info, "base/add_test.c:100");
+  REQUIRE (ret != -1);
+  REQUIRE (test_server_info.stat_info[ret].second == allocated_size_2);
+}
+
+// Test add_stat() w/ abnormal case
+//
+// Test add_stat() with single thread when different file paths
+// that will be made same stat name are given (abnormal case)
+// There is a potential issue scenario that can't distinguish different allocation. (TODO)
```

Comment:
How about write a function name ? 

```suggestion
// There is a potential issue scenario that can't distinguish different allocation. (TODO in memory_monitor::make_stat_name ())
```